src/parser/Literal.cpp

```diff
@@ -149,4 +149,19 @@ void Literal::setSubstr(std::size_t start, std::size_t length) {
 // __________________________________________
 void Literal::removeDatatypeOrLanguageTag() { content_.erase(beginOfSuffix_); }
 
+// __________________________________________
+void Literal::replaceContent(const std::string& newContent) {
+  std::size_t minLength = std::min(beginOfSuffix_ - 2, newContent.size());
+  for (std::size_t i = 0; i < minLength; ++i) {
+    content_[i + 1] = newContent[i];
+  }
+  if (newContent.size() <= beginOfSuffix_ - 2) {
```

Comment:
at the beginning, assign the result of `beginOfsuffix_ -2` to a variable `originalContentLength`, which makes these calculations much more readable.